File: python/janus/providers/anthropic.py
"""Anthropic provider implementation."""

import os

from anthropic import AsyncAnthropic

from ..core.types import ProviderConfig
from .base import CompletionOptions, Provider


class AnthropicProvider(Provider):
    """Provider for Anthropic's Claude API."""

    def __init__(self, config: ProviderConfig):
        super().__init__("Anthropic")
        self.model = config.model or "claude-opus-4-5-20251101"
        self.client = AsyncAnthropic(
            api_key=config.api_key or os.environ.get("ANTHROPIC_API_KEY")
        )

    async def generate(self, prompt: str, options: CompletionOptions | None = None) -> str:
        """Generate a completion using Anthropic's API."""
        options = options or CompletionOptions()

        try:
            kwargs = {
                "model": self.model,
                "max_tokens": options.max_tokens,
                "messages": [{"role": "user", "content": prompt}],
            }

            if options.system_prompt:
                kwargs["system"] = options.system_prompt

            if options.temperature is not None:
                kwargs["temperature"] = options.temperature

            response = await self.client.messages.create(**kwargs)

            # Extract text from response
            return "".join(
                block.text for block in response.content if hasattr(block, "text")
            )
        except Exception as e:
            return f"[Error] Anthropic failed to generate response: {e}"

File: conclave_cli/src/providers/anthropic.ts
import Anthropic from '@anthropic-ai/sdk';
import { Provider, CompletionOptions } from './base.js';
import { ProviderConfig } from '../core/types.js';

export class AnthropicProvider implements Provider {
    public name = 'Anthropic';
    private client: Anthropic;
    private model: string;

    constructor(config: ProviderConfig) {
        this.model = config.model || 'claude-3-5-sonnet-20240620';
        this.client = new Anthropic({
            apiKey: config.api_key || process.env.ANTHROPIC_API_KEY,
        });
    }

    async generate(prompt: string, options?: CompletionOptions): Promise<string> {
        try {
            const msg = await this.client.messages.create({
                model: this.model,
                max_tokens: options?.maxTokens || 4096,
                temperature: options?.temperature ?? 0.7,
                system: options?.systemPrompt || 'You are a helpful assistant.',
                messages: [{ role: 'user', content: prompt }],
            });

            // Anthropic content can be blocks, we extract text
            const textBlock = msg.content.find(c => c.type === 'text');
            return textBlock?.text || '';
        } catch (error) {
            console.error(`Anthropic Error:`, error);
            return `[Error] Anthropic failed to generate response: ${error}`;
        }
    }
}
